src/components/TextSlide/TextSlide.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import Button from '@material-ui/core/Button';
+import Slide from '@material-ui/core/Slide';
+import Grid from '@material-ui/core/Grid';
+import CountUp from 'react-countup';
+
+export default function TextSlide(props) {
+  const passwordGuesser = (
+    (props.usesInput !== true) ? null :
+    (props.inputType === 'num') ? (
+      <CountUp
+        start={0}
+        duration={1/10000 * parseInt(props.userInput, 10)}
+        end={parseInt(props.userInput, 10)}
+        formattingFn={num => String(num).padStart(props.inputLength, '0')}
+        useEasing={false}>
+        {({ countUpRef, start }) => (
+          <div>
+            <span ref={countUpRef} />
+            <Button onClick={start}>Start</Button>
+          </div>
+        )}
+      </CountUp>) : null
```

Comment:
This custom component only supports numbers, right? So refactoring for letters will be harder?

Reply:
Yeah, it only supports numbers. I was thinking I could create a custom formatting function that would basically convert integers into strings, i.e. 1 = "a", 2 = "b", ..., 26 = "z", 27 = "aa", and so on. I don't think it will be too hard, and then we can still use the CountUp module for the letter parts. LMK what you think!